### ISSUE TITLE
`<mask>` area doesn't get expanded beyond object's bounding box

### ISSUE THREAD

**Issue by stanio** [weisJ/jsvg#65]
It appears `<mask>` area doesn't get expanded beyond the bounding box of the object it is applied to:

-   [mask-0.svg](https://github.com/weisJ/jsvg/assets/1247730/9141bf66-afbc-4130-b381-55fc0865b8fe), [mask-1.svg](https://github.com/weisJ/jsvg/assets/1247730/419a0a0c-59b9-4d28-a50d-5e7cc931f7c0), [mask-2.svg](https://github.com/weisJ/jsvg/assets/1247730/cdb62019-b420-4d90-9c6c-98bb5956f5a3)

It could "shrink" inside the object's bounding box, though.  The most apparent effect is that a stroke that goes beyond the object's bounding box gets clipped from view.

`mask-0.svg`

```xml
    <mask id="aperture" maskUnits="userSpaceOnUse"
        x="0" y="0" width="500" height="500">
```

Actual | Expected
---|---
<img alt="jsvg-mask-0" src="https://github.com/weisJ/jsvg/assets/1247730/bb77bf65-67a4-47cd-ade7-12f52df5b3b1" width="252"> | <img alt="chrome-mask-0" src="https://github.com/weisJ/jsvg/assets/1247730/ce15f469-a4e5-4a6e-add3-7f3955ca2637" width="252">

`mask-1.svg`

```xml
    <mask id="aperture" maskUnits="userSpaceOnUse"
        x="50" y="50" width="450" height="450">
```

Actual | Expected
---|---
<img alt="jsvg-mask-1" src="https://github.com/weisJ/jsvg/assets/1247730/96af702a-58c9-496a-994b-cc7ba606a7eb" width="252"> | <img alt="chrome-mask-1" src="https://github.com/weisJ/jsvg/assets/1247730/6836d7df-5865-4d02-ae23-914a24735bb9" width="252">

`mask-2.svg`

```xml
    <mask id="aperture" maskUnits="userSpaceOnUse"
        x="60" y="60" width="440" height="440">
```

Actual | Expected
---|---
<img alt="jsvg-mask-2" src="https://github.com/weisJ/jsvg/assets/1247730/50a7d6d3-8ea8-4782-b543-4d4ff42f1576" width="252"> | <img alt="chrome-mask-2" src="https://github.com/weisJ/jsvg/assets/1247730/b970b2e1-7f84-4eba-b3a2-1c34fc40a500" width="252">

----

Initially found in https://github.com/weisJ/jsvg/issues/64#issuecomment-1974806550

**Comment by weisJ** [comment]
This should be fixed in the latest snapshot

**Comment by stanio** [comment]
Trying out current [1.5.0-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/com/github/weisj/jsvg/1.5.0-SNAPSHOT/) (`gradlew --refresh-dependencies` → `1.5.0-20240304.002615-19`) – the outer stroke is now fully visible but a different defect appears.  Looks like the overall mask is scaled (enlarged):

mask-0 | mask-1 | mask-2
---|---|---
<img alt="mask-0" src="https://github.com/weisJ/jsvg/assets/1247730/b1d13a79-91d6-44e4-9af0-c341b00701ba" width="252"> | <img alt="mask-1" src="https://github.com/weisJ/jsvg/assets/1247730/3fae71ae-41b5-4563-9a35-f6ce7d841a25" width="252"> | <img alt="mask-2" src="https://github.com/weisJ/jsvg/assets/1247730/3359d5fa-2cc5-440e-855a-b567d16b0dcd" width="252">

Notice the circular hole is bigger than the debug overlay that uses the same path as for the mask.

**Comment by weisJ** [comment]
This is related to #66. I have a fix locally, will push once #67 is resolved. I noticed that the library I am using to compare images in tests doesn’t handle transparent pixels and thus multiple tests have failed silently for the past few versions. I need to catch up with those before work on subpixel alignment can be continued.

**Comment by stanio** [comment]
Just tried with a current 1.5.0-SNAPSHOT – it now (currently) appears just right.
